internal/operands/vm-delete-protection/reconcile_test.go

```diff
@@ -0,0 +1,151 @@
+package vm_delete_protection
+
+import (
+	"context"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	ssp "kubevirt.io/ssp-operator/api/v1beta2"
+	"kubevirt.io/ssp-operator/internal/common"
+	. "kubevirt.io/ssp-operator/internal/test-utils"
+)
+
+var log = logf.Log.WithName("vm_delete_protection_operand")
+
+var _ = Describe("VM delete protection operand", func() {
+	const (
+		namespace = "kubevirt"
+		name      = "test-ssp"
+	)
+
+	var (
+		request common.Request
+		operand = New()
+	)
+
+	BeforeEach(func() {
+		client := fake.NewClientBuilder().WithScheme(common.Scheme).Build()
+		request = common.Request{
+			Request: reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: namespace,
+					Name:      name,
+				},
+			},
+			Client:  client,
+			Context: context.Background(),
+			Instance: &ssp.SSP{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "SSP",
+					APIVersion: ssp.GroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+			},
+			Logger:       log,
+			VersionCache: common.VersionCache{},
+		}
+	})
+
+	It("should create VM deletion protection resources", func() {
+		_, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		ExpectResourceExists(NewVMDeleteProtectionValidatingAdmissionPolicy(), request)
+		ExpectResourceExists(NewVMDeleteProtectionValidatingAdmissionPolicyBinding(), request)
+	})
+
+	DescribeTable("should update labels if changed", func(obj client.Object, objName string) {
```

Comment:
These tests are incorrect. SSP should not remove user defined labels and annotations.
This functionality is partially covered in the tests for `internal/common`: https://github.com/kubevirt/ssp-operator/blob/f8a7389aa787fb1712fe54c63b09848f33ac208c/internal/common/resource_test.go#L72

Reply:
I've dropped the test, is there anything additional that we should cover in a separate test about labels in your opinion?

Reply:
No. IIUC there is no special behavior for labels in VAP or VAPB.